### ISSUE TITLE
Version 1.8.3 doesn't contain Revert "commit: fix const declaration"

### ISSUE THREAD

**Issue by carrotIndustries** [libgit2/libgit2#6934]
### Reproduction steps

Build a program that expects that all versions >= 1.8.2 will have the pre-1.8.0 constness of `git_commit_create` and friends.

See https://github.com/horizon-eda/horizon/commit/3b6b783db797b593119756821c04ac64b66b7ffa

### Expected behavior

Build passes.

### Actual behavior

Build fails: https://github.com/horizon-eda/horizon/actions/runs/11576015213/job/32223990024?pr=789

### Version of libgit2 (release number or SHA1)

1.8.3

### Operating system(s) tested

Arch Linux

**Comment by carrotIndustries** [comment]
For those looking for a C++ fix that can deal with the changing constness: https://github.com/horizon-eda/horizon/commit/d5a46d2d7e879e82d0d2798715c1369f8ae54fcd

**Comment by ethomson** [comment]
Sheesh, what a mess.

**Comment by carrotIndustries** [comment]
What struck me as a bit odd is that this change doesn't show up when comparing 1.8.3 against 1.8.2: https://github.com/libgit2/libgit2/compare/v1.8.2...v1.8.3

**Comment by ethomson** [comment]
We did a release candidate for v1.8.2, which is something that we never do. We don't really have a defined process for that, and it meant that we "shipped" v1.8.2 without actually updating the `maint/v1.8` branch. Which meant that we clobbered it when we shipped v1.8.3 which _did_ land in `maint/v1.8`. 😭 

My bad, sloppy release engineering. I will spin up a process for release candidates so that we don't have this happen again.
